bin/node/cli/src/service.rs

```diff
@@ -249,38 +241,9 @@ macro_rules! new_full {
 	}}
 }
 
-type ConcreteBlock = node_primitives::Block;
-type ConcreteClient =
-	Client<
-		Backend<ConcreteBlock>,
-		LocalCallExecutor<Backend<ConcreteBlock>, NativeExecutor<node_executor::Executor>>,
-		ConcreteBlock,
-		node_runtime::RuntimeApi
-	>;
-type ConcreteBackend = Backend<ConcreteBlock>;
-type ConcreteTransactionPool = sc_transaction_pool::BasicPool<
-	sc_transaction_pool::FullChainApi<ConcreteClient, ConcreteBlock>,
-	ConcreteBlock
->;
-
```

Comment:
There were reasons for this code, it should not be removed.

Reply:
So the whole point of this PR is to make the `Client` struct internal to the substrate codebase. The above piece of code could not exist with this idea in mind, I've compiled it and it works fine.

Maybe you could clarify what other reasons it needs to exist and we can work around it

Reply:
https://github.com/paritytech/substrate/pull/3709

And the `Client` struct is still public, it appears in the API of `AbstractService`.

Reply:
Ooh, this complicates things a bit

Reply:
If you remove the client completely from the public api, we can probably solve this differently, but currently you still have `Client` in `AbstractService`.

Reply:
so I made `Client` in `AbstractSerivice` an associated type, which implement all the traits client does here: https://github.com/paritytech/substrate/blob/743ce140a638cdcb586e60396af6f0e4b626b53a/client/service/src/lib.rs#L149-L179